parsec/core/cli/invitation.py

```diff
@@ -0,0 +1,395 @@
+        else:
+            assert invitation["type"] == InvitationType.DEVICE
+            initial_ctx = DeviceGreetInitialCtx(cmds=cmds, token=token)
+            do_greet = partial(_do_greet_device, device, initial_ctx)
+
+        while True:
+            try:
+                greet_done = await do_greet()
+                if greet_done:
+                    break
+            except InviteError as exc:
+                click.secho(str(exc), fg="red")
+            click.secho("Restarting the invitation process", fg="red")
+
+
+@click.command(short_help="greet invitation")
+@core_config_and_device_options
+@click.argument("token", type=UUID)
+def greet_invitation(config, device, token, **kwargs):
+    """
+    Greet a new device or user into the organization
+    """
+    with cli_exception_handler(config.debug):
+        # Disable task monitoring given user prompt will block the coroutine
+        trio_run(_greet_invitation, config, device, token, monitor_tasks=False)
+
+
+async def _do_claim_user(initial_ctx):
+    async with spinner("Initializing connection with greeter for claiming user"):
+        in_progress_ctx = await initial_ctx.do_wait_peer()
+
+    choices = in_progress_ctx.generate_greeter_sas_choices(size=3)
+    for i, choice in enumerate(choices):
+        display_choice = click.style(str(choice), fg="yellow")
+        click.echo(f" {i} - {display_choice}")
+    code = click.prompt(
+        f"Select code provided by greeter", type=click.Choice([str(x) for x in range(len(choices))])
+    )
+    if choices[int(code)] != in_progress_ctx.greeter_sas:
+        click.secho("Wrong code provided", fg="red")
+        return None
+
+    in_progress_ctx = await in_progress_ctx.do_signify_trust()
+    display_claimer_sas = click.style(str(in_progress_ctx.claimer_sas), fg="yellow")
+    click.echo(f"Code to provide to greeter: {display_claimer_sas}")
+    async with spinner("Waiting for greeter"):
+        in_progress_ctx = await in_progress_ctx.do_wait_peer_trust()
+
+    requested_label = click.prompt("User fullname")
+    requested_email = initial_ctx.claimer_email
+    default_device_id = f"{requested_email.split('@', 1)[0]}@{platform.node()}"
+    requested_device_id = click.prompt("Device ID", default=default_device_id, type=DeviceID)
+    async with spinner("Waiting for greeter (finalizing)"):
+        new_device = await in_progress_ctx.do_claim_user(
+            requested_device_id=requested_device_id,
+            requested_human_handle=HumanHandle(email=requested_email, label=requested_label),
+        )
+
+    return new_device
+
+
+async def _do_claim_device(initial_ctx):
+    async with spinner("Initializing connection with greeter for claiming device"):
+        in_progress_ctx = await initial_ctx.do_wait_peer()
+
+    choices = in_progress_ctx.generate_greeter_sas_choices(size=3)
+    for i, choice in enumerate(choices):
+        display_choice = click.style(str(choice), fg="yellow")
```

Comment:
> `str(choice)`

So we're not using the 32-symbol alphabet here?
```python
In [1]: dict(enumerate("ABCDEFGHJKLMNPQRSTUVWXYZ23456789"))
Out[1]: 
{0: 'A',
 1: 'B',
 2: 'C',
 3: 'D',
 4: 'E',
 5: 'F',
 6: 'G',
 7: 'H',
 8: 'J',
 9: 'K',
 10: 'L',
 11: 'M',
 12: 'N',
 13: 'P',
 14: 'Q',
 15: 'R',
 16: 'S',
 17: 'T',
 18: 'U',
 19: 'V',
 20: 'W',
 21: 'X',
 22: 'Y',
 23: 'Z',
 24: '2',
 25: '3',
 26: '4',
 27: '5',
 28: '6',
 29: '7',
 30: '8',
 31: '9'}
```

Reply:
I found the 7 digits number code both simple to read and the easiest to maintain

I guess the advantage of your technique is to reduce the size of the code (only need 4 characters), @Max-7 preference on this ?

(obviously if we change this, we should modify `generate_greeter_sas_choices` so that it returns a 4 characters string, hence the complexity is hidden for the CLI/GUI)

Reply:
I'm all for 4 symbols from a 32-symbols alphabet. Compare the following:
```
C9VC : 0085986
QZKW : 0665326
XCUU : 0608341
X4ZB : 0057173
Y866 : 0947158
NVGG : 0203372
T5J9 : 1024881
MHSW : 0671979
TGKJ : 0271569
RNNQ : 0471439
```

Reply:
👍